pkg/trtl/metrics.go

```diff
@@ -0,0 +1,141 @@
+package trtl
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/zerolog/log"
+)
+
+var (
+	pmPuts  *prometheus.CounterVec // count of trtl Puts per namespace
+	pmGets  *prometheus.CounterVec // count of trtl Gets per namespace
+	pmDels  *prometheus.CounterVec // count of trtl Deletes per namespace
+	pmIters *prometheus.CounterVec // count of trtl Iters per namespace
+	// pmObjects    *prometheus.CounterVec   // count of objects being managed by trtl, by namespace
+	// pmTombstones *prometheus.CounterVec   // count of tombstones per namespace; increases on delete, decrease on overwrite of tombstone
+	pmLatency *prometheus.HistogramVec // the time it is taking for successful RPC calls to complete, labeled by RPC type, success, and failure
+)
+
+// A MetricsService manages Prometheus metrics
+type MetricsService struct {
+	srv *http.Server
+}
+
+func NewMetricsService() (*MetricsService, error) {
+	return &MetricsService{}, nil
+}
+
+// Serve serves the Prometheus metrics
+func (m *MetricsService) Serve(addr string) error {
+	if err := registerMetrics(); err != nil {
+		return err
+	}
+	m.srv = serveMetrics(addr)
+
+	return nil
+}
+
+// Gracefully shutdown the Prometheus metrics service
+func (m *MetricsService) Shutdown() error {
+	// Might want to share context from Trtl more globally?
+	ctx := context.Background()
+	if err := m.srv.Shutdown(ctx); err != nil {
+		log.Error().Err(err).Msg("unable to gracefully shutdown prometheus metrics")
+	}
+	return nil
+}
```

Comment:
Wasn't sure this was quite what you had in mind for `Serve` and `Shutdown` @bbengfort so I guessed; happy to redo this, just let me know what you were thinking.